utils/fs/memory/memory.go

```diff
@@ -89,26 +90,25 @@ func (fs *Memory) ReadDir(base string) (entries []fs.FileInfo, err error) {
 		fullpath, _ = filepath.Rel(base, fullpath)
 		parts := strings.Split(fullpath, string(separator))
 
-		if len(parts) != 1 {
-			dirs[parts[0]] = true
+		if len(parts) == 1 {
+			entries = append(entries, newFileInfo(fs.base, fullpath, f.content.size))
 			continue
 		}
 
-		entries = append(entries, newFileInfo(fs.base, fullpath, f.c.size))
-	}
+		if _, ok := appendedDirs[parts[0]]; ok {
+			continue
+		}
 
-	for path := range dirs {
-		entries = append(entries, &fileInfo{
-			name:  path,
-			isDir: true,
-		})
+		entries = append(entries, &fileInfo{name: parts[0], isDir: true})
+		appendedDirs[parts[0]] = true
 	}
 
 	return
 }
 
 func (fs *Memory) TempFile(dir, prefix string) (fs.File, error) {
-	filename := fmt.Sprintf("%s_%d", prefix, time.Now().UnixNano())
+	fs.tempCount++
+	filename := fmt.Sprintf("%s_%d_%d", prefix, fs.tempCount, time.Now().UnixNano())
```

Comment:
This does not solve the problem.

What is the purpose of TempFile()? If its purpose is to create a new file, then a new file has to be created, instead of truncating a file with a highly uncommon name. Also explain what this method is supposed to do and why it works differently than the standard TempFile method.